### PR TITLE
ErrorDialog Pop-Up

### PR DESCRIPTION
**Pull request by zzori-theoriginal**
The 1st part of the ticket has been completed, with the error window now using the native functionality of Visual Studio.

After this, I will be adding the theming fixes to this PR.

### CODE REVIEW

File: src/Services/src/Dialog/IDialogService.cs [hunk 1]
```diff
@@ -7,6 +7,6 @@ public interface IDialogService
         IDialogResult ShowDialog(string dialogViewModel, object parameter = null);
 
         void CloseDialog(object dialogWindow, bool result);
-        void DisplayErrorDialog(string errorTitle, string errorMsg);
+
```
Comment (ccheetham) [inline src/Services/src/Dialog/IDialogService.cs:10]: remove blank line

File: src/ViewModels/src/CloudExplorer/CloudExplorerViewModel.cs [hunk 2]
```diff
@@ -15,6 +17,7 @@ public class CloudExplorerViewModel : AbstractViewModel, ICloudExplorerViewModel
         internal static readonly string _stopAppErrorMsg = "Encountered an error while stopping app";
         internal static readonly string _startAppErrorMsg = "Encountered an error while starting app";
         internal static readonly string _deleteAppErrorMsg = "Encountered an error while deleting app";
+        private static IErrorDialog _dialogService;
```
Comment (ccheetham) [inline src/ViewModels/src/CloudExplorer/CloudExplorerViewModel.cs:20]: You've scoped this field as `static` so there's only ever one instance, yet you initialize in constructor.  So if you construct 2 or more of these objects, each will overwrite the previous service instance.

File: src/ViewModels/src/DeploymentDialog/DeploymentDialogViewModel.cs [hunk 3]
```diff
@@ -25,6 +27,7 @@ public class DeploymentDialogViewModel : AbstractViewModel, IDeploymentDialogVie
         private string _status;
         private string _appName;
         private readonly bool _fullFrameworkDeployment = false;
+        private static IErrorDialog _dialogService;
```
Comment (ccheetham) [inline src/ViewModels/src/DeploymentDialog/DeploymentDialogViewModel.cs:30]: same issue as above

File: src/ViewModels/src/TreeViewItems/CfInstanceViewModel.cs [hunk 4]
```diff
@@ -12,12 +14,14 @@ public class CfInstanceViewModel : TreeViewItemViewModel
         internal static readonly string _emptyOrgsPlaceholderMsg = "No orgs";
         internal static readonly string _loadingMsg = "Loading orgs...";
         internal static readonly string _getOrgsFailureMsg = "Unable to load orgs";
+        private static IErrorDialog _dialogService;
```
Comment (ccheetham) [inline src/ViewModels/src/TreeViewItems/CfInstanceViewModel.cs:17]: beginning to see a pattern :)

**Review by ccheetham — CHANGES_REQUESTED** [review]
there are several suspect instances of using a static field in a constructor.  There may be more than just the few I tagged.

File: src/ViewModels/src/TreeViewItems/CfInstanceViewModel.cs (hunk 4 above)
Comment (zzori-theoriginal) [reply src/ViewModels/src/TreeViewItems/CfInstanceViewModel.cs:17]: saving this and all the above changes for a new issue that will cover changing the use of the `static` keyword in all files, where necessary

File: src/Services/src/Dialog/IDialogService.cs (hunk 1 above)
Comment (zzori-theoriginal) [reply src/Services/src/Dialog/IDialogService.cs:10]: done

File: src/ViewModels/src/TreeViewItems/CfInstanceViewModel.cs (hunk 4 above)
Comment (zzori-theoriginal) [reply src/ViewModels/src/TreeViewItems/CfInstanceViewModel.cs:17]: @jkonicki I don't mind doing all the other fixes in this ticket as well if that is what you would like. I just didn't know if for the sake of keeping things organized, we should split it up.

Comment (zzori-theoriginal) [reply src/ViewModels/src/TreeViewItems/CfInstanceViewModel.cs:17]: but I will go ahead and make the change that chris mentioned here

Comment (zzori-theoriginal) [reply src/ViewModels/src/TreeViewItems/CfInstanceViewModel.cs:17]: Update: I went ahead and started the process of getting rid of the `static` modifier in all files across al projects. so far, it has been fine, but if changing the keywords does cause any code related issues, I will take note of it and see what other options we have.

Comment (zzori-theoriginal) [reply src/ViewModels/src/TreeViewItems/CfInstanceViewModel.cs:17]: Update #2: After talking with Chris, it was determined that _some_ of the Singletons that are being instantiated in the project should actually keep the `static` keyword( i.e. the `Commands` files in the VS project). Therefore, I am shifting back to only focusing on removing the `static` keyword from `Service` instances.

**Review by ccheetham — DISMISSED** [review]
lgtm

**Review by dtillman — DISMISSED** [review]
LGTM

**Review by ccheetham — APPROVED** [review]
lgtm

**Pull request closed without merging**